packages/mobx-state-tree/src/types/primitives.ts

```diff
@@ -116,6 +118,44 @@ export const integer: ISimpleType<number> = new CoreType<number, number, number>
     (v) => isInteger(v)
 )
 
+/**
+ * `types.float` - Creates a type that can only contain an float value.
+ * This type is used for float values by default
+ *
+ * Example:
+ * ```ts
+ * const Size = types.model({
+ *   width: types.float,
+ *   height: 10
+ * })
+ * ```
+ */
+// tslint:disable-next-line:variable-name
+export const float: ISimpleType<number> = new CoreType<number, number, number>(
+    "float",
+    TypeFlags.Float,
+    (v) => isFloat(v)
+)
+
+/**
+ * `types.finite` - Creates a type that can only contain an finite value.
+ * This type is used for finite values by default
```

Comment:
nitpick: @chakrihacker - this is not technically true, I don't think we infer types.finite with this change. I think this comment is true of types.number, based on how [getPrimitiveFactoryFromValue behaves](https://github.com/mobxjs/mobx-state-tree/blob/85445de5f0e90d8c30545dfa008e2704d1eaebbb/packages/mobx-state-tree/src/types/primitives.ts#L224), which is what we call in [toPropertiesObject](https://github.com/mobxjs/mobx-state-tree/blob/85445de5f0e90d8c30545dfa008e2704d1eaebbb/packages/mobx-state-tree/src/types/complex-types/model.ts#L258) when doing type inference.

I'm going to make this edit since it's small and really more of a nitpick than anything.